src/components/SearchFilter/SearchFilter.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { useContext } from "react";
+import { useForm } from "react-hook-form";
+import { useHistory } from "react-router-dom";
+import { Select } from "../Select/Select";
+import { DataList } from "../DataList/DataList";
+import { SERVICES, DATALIST, PETS } from "../../constants/SearchOptions";
+import AnnouncementsContext from "../../context/announcements/announcements-context";
```

Comment:
About the proper way of creating and using contexts, I send you back to the Kent article I sent you before.